.test-infra/mock-apis/src/main/go/internal/service/echo/echo.go

```diff
@@ -0,0 +1,185 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package echo contains the EchoService API implementation.
+package echo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"path"
+	"reflect"
+	"time"
+
+	"github.com/apache/beam/test-infra/mock-apis/src/main/go/internal/cache"
+	"github.com/apache/beam/test-infra/mock-apis/src/main/go/internal/logging"
+	"github.com/apache/beam/test-infra/mock-apis/src/main/go/internal/metric"
+	echov1 "github.com/apache/beam/test-infra/mock-apis/src/main/go/internal/proto/echo/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	metricsNamePrefix = "echo"
+	echoPath          = "/proto.echo.v1.EchoService/Echo"
+	echoPathAlias     = "/v1/echo"
+	healthPath        = "/grpc.health.v1.Health/Check"
+	healthPathAlias   = "/v1/healthz"
+)
+
+type Options struct {
+	Decrementer   cache.Decrementer
+	MetricsWriter metric.Writer
+	Logger        *slog.Logger
+	LoggingAttrs  []slog.Attr
+}
+
+// Register a grpc.Server with the echov1.EchoService. Returns a http.Handler or error.
+func Register(s *grpc.Server, opts *Options) (http.Handler, error) {
+	if opts.Logger == nil {
+		opts.Logger = logging.New(&logging.Options{
```

Comment:
Same comment here WRT the logger and the always used LoggingAttrs. Use `.With()` to build them into the logger.

Reply:
You mentioned not to undo work in the context of slog. Does your comment apply in this context as well? FYI, I will be doing the infrastructure as code PR and may uncover a couple of things in the code that might need fixing so perhaps I can refactor the slog related stuff there.